back/scripts/datasets/datafile_loader.py

```diff
@@ -37,40 +39,26 @@ def __init__(self, communities_selector, topic_config):
         # Drop duplicates and cast data to schema types
         self.normalized_data = self._normalize_data()
 
-    # Internal function to load JSON schema from URL
     def _load_schema(self, schema_topic_config):
-        # Load JSON schema from URL
-        json_schema_loader = BaseLoader.loader_factory(schema_topic_config["url"])
-        json_schema = json_schema_loader.load()
-        # Flatten JSON schema to DataFrame
-        schema_name = schema_topic_config["name"]
-        flattened_schema = flatten_json_schema(json_schema, schema_name)
-        # Convert flattened schema to DataFrame
-        schema_df = pd.DataFrame(flattened_schema)
-        # In "type" column, replace NaN values by "string" (default value)
-        schema_df["type"].fillna("string", inplace=True)
-        return schema_df
-
-    # Load data from URL and flatten it to DataFrame
+        self.official_schema = MarchesPublicsSchemaLoader.load(
```

Comment:
This specific json was not dataframe compatible. There was already a dedicated logic that I moved to a dedicated class at the bottom.